crates/ruff_cli/src/lib.rs

```diff
@@ -117,11 +118,34 @@ quoting the executed command, along with the relevant file contents and `pyproje
             shell.generate(&mut Args::command(), &mut io::stdout());
         }
         Command::Check(args) => return check(args, log_level),
+        Command::Format { files } => return format(&files),
     }
 
     Ok(ExitStatus::Success)
 }
 
+#[cfg_attr(not(feature = "format"), allow(unused))]
+fn format(files: &[PathBuf]) -> Result<ExitStatus> {
+    // dummy
+    let format_code = |code: &str| code.replace("# DEL", "");
+
+    let is_stdin = files == vec![PathBuf::from("-")];
```

Comment:
Nit: This equals check performs, if Rust isn't able to optimize it away, two allocations. One for the `vec` and another for the `PathBuf`. I assume you plan to rework this in follow-up PRs?

Reply:
I copied this from https://github.com/charliermarsh/ruff/blob/7cae4eaf086e5b952fa52a1d94678985f710f58a/crates/ruff_cli/src/lib.rs#L312

I would have hoped that llvm can see through this and just optimizes this away, but if not i can rewrite it

Reply:
One thing we could do that is low effort and also avoids ambiguity (what if I call ruff with `ruff format -` because I have a `-` directory) is to introduce a new `enum` with a `Stdin { filename }` and `Paths(Vec<PathBuf>)` variants.

Reply:
i've changed this to a no-alloc version